spec/controllers/adjustments_controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+RSpec.describe AdjustmentsController, type: :controller do
+  let(:organization) { create(:organization) }
+  let(:user) { create(:user, organization: organization) }
+  let(:storage_location) { create(:storage_location, organization: organization) }
+  let(:item1) { create(:item, name: "Item One", organization: organization) }
+  let(:item2) { create(:item, name: "Item Two", organization: organization) }
+
+  let!(:adjustment1) do
+    adj = create(:adjustment,
+      organization: organization,
+      storage_location: storage_location,
+      comment: "First adjustment",
+      created_at: 1.day.ago)
+    adj.line_items << build(:line_item, quantity: 10, item: item1, itemizable: adj)
+    adj.line_items << build(:line_item, quantity: 5, item: item2, itemizable: adj)
+    adj
+  end
+
+  let!(:adjustment2) do
+    adj = create(:adjustment,
+      organization: organization,
+      storage_location: storage_location,
+      comment: "Second adjustment",
+      created_at: 5.days.ago)
+    adj.line_items << build(:line_item, quantity: -5, item: item1, itemizable: adj)
+    adj
+  end
+
+  before do
+    sign_in(user)
+  end
+
+  describe "GET #index" do
+    context "with CSV format" do
+      it "returns a CSV file" do
+        get :index, format: :csv
+        expect(response).to be_successful
+        expect(response.header["Content-Type"]).to include "text/csv"
+      end
+
+      it "includes appropriate headers for adjustments" do
+        get :index, format: :csv
+        expect(response.body).to include("Created date")
```

Comment:
I usually prefer comparing the CSV against actual values - see for example [https://github.com/rubyforgood/human-essentials/blob/d8fa236212a4114d962d4f7ac1002f42ba058c2e/spec/requests/items_requests_spec.rb#L52-L52](https://github.com/rubyforgood/human-essentials/blob/d8fa236212a4114d962d4f7ac1002f42ba058c2e/spec/requests/items_requests_spec.rb#L52-L52) .

If the data you're using is too dynamic to do this, that's a good indication that your data's too dynamic. :smile: It's better to hardcode data to ensure confidence that things aren't passing by "accident" (which is absolutely something that has happened in the past).

Reply:
good point, I will change to more static direct csv-to-csv string assertion, thanks :)